sfxclient/sfxclient.go

```diff
@@ -220,6 +223,11 @@ func (s *Scheduler) CollectDatapoints() []*datapoint.Datapoint {
 		for _, p := range s.callbackMap {
 			ret = append(ret, p.getDatapoints(now, s.SendZeroTime)...)
 		}
+		if gatewayMetricPrefix == s.prefix {
+			for _, datapoint := range ret {
+				datapoint.Metric = gatewayMetricPrefix + datapoint.Metric
```

Comment:
this is insanely expensive operation and given the fact that we run CollectDatapoints() i would advise not to use "+" operation

Reply:
why are we not appending the prefix at the sender? golib should not mingle around with metric prefix name.

Reply:
There are multiple sources which implement this interface at the sender level. We will have to add this logic for each sender.